developer_guides/rimage/ipc4_extended_module_config.rst

```diff
@@ -0,0 +1,219 @@
+.. _ipc4_extended_module_config:
+
+Ipc4_extended_module_config
+###########################
+
+The ipc4 extended module config is stored in the extended manifest structure of the
+output binary file. This config includes many module attributes such as name, uuid
+or capabilities. The host driver will rely on the extended module config for IPC
+messages to the DSP
+
+Rimage builds extended module config based on module setting in config/xxx.toml.
+The following setting is for gain module.
+
+.. code-block:: bash
+
+        [[module.entry]]
+        name = "GAIN"
+        uuid = "61BCA9A8-18D0-4A18-8E7B-2639219804B7"
+        affinity_mask = "0x1"
+        instance_count = "40"
+        domain_types = "0"
+        load_type = "0"
+        module_type = "5"
+        auto_start = "0"
+
+        # pin = [dir, type, sample rate, size, container, channel-cfg]
+        pin = [0, 0, 0xfeef, 0xf, 0xf, 0x45ff,
+                        1, 0, 0xfeef, 0xf, 0xf, 0x1ff]
+
+        # mod_cfg [PAR_0 PAR_1 PAR_2 PAR_3 IS_BYTES CPS IBS OBS MOD_FLAGS CPC OBLS]
+        mod_cfg = [0, 0, 0, 0, 416, 914000, 48, 64, 0, 0, 0,
+                   1, 0, 0, 0, 416, 1321600, 192, 256, 0, 0, 0,
+                   2, 0, 0, 0, 416, 1786000, 192, 256, 0, 0, 0,
+                   3, 0, 0, 0, 416, 2333000, 48, 64, 0, 0, 0,
+                   4, 0, 0, 0, 416, 2910000, 192, 256, 0, 0, 0,
+                   5, 0, 0, 0, 416, 3441000, 192, 256, 0, 0, 0,
+                   6, 0, 0, 0, 416, 4265000, 192, 256, 0, 0, 0]
+
+Module entry definition
+***********************
+| **name**: name of the module
+| **uuid**: uuid of the module
+| **affinity_mask**: restriction on which core a module can run, with the actual determination done on a per-instance basis.
+|                    bit N is for core N, 0x3 means module instance can run on core 0 and 1
+| **instance_count**: max module instance count can be active in fw
```

Comment:
same here, what is "don't care"

Reply:
The problem is that we have limited dsp memory so we can't have unlimited instance count. The instance count is used for avoiding out of memory. You can give a huge number to remove the limitation literally.

Reply:
Yea but we also have no comprehension of the DSP we are running on. Not all DSPs are the same.